spec/support/graphql_helper.rb

```diff
@@ -8,17 +8,30 @@ def graphql(query, variables = {})
     post '/graphql', params: params
   end
 
-  def projects_query
-    <<~GQL
-      query {
-        projects {
-          stringKey
-          displayLabel
-          projectUrl
-          isPrimary
+  def projects_query(is_primary: nil)
+    if is_primary.nil?
```

Comment:
```suggestion
    is_primary_filter = is_primary ? "isPrimary: \"#{is_primary}\"" : ""
    <<~GQL
      query {
         projects(#{is_primary_filter}) {
         .............
```

Reply:
This way you don't have to have the query written out twice. I think you can also use graphql variables.

Reply:
Going with `projects(isPrimary: #{is_primary})` in the end (as we just talked about), since apparently passing null is fine here.  Thanks for suggesting the alternative!  Much better than what I had originally.